spec/models/manageiq/providers/vmware/infra_manager/event_catcher/runner_spec.rb

```diff
@@ -0,0 +1,115 @@
+require "spec_helper"
+
+describe ManageIQ::Providers::Vmware::InfraManager::EventCatcher::Runner do
+  let(:ems)      { FactoryGirl.create(:ems_vmware, :hostname => "hostname") }
+  let(:catcher)  { ManageIQ::Providers::Vmware::InfraManager::EventCatcher::Runner.new(:ems_id => ems.id) }
+  let(:settings) { {:flooding_monitor_enabled => false} }
+
+  before do
+    allow_any_instance_of(ManageIQ::Providers::Vmware::InfraManager).to receive(:authentication_check).and_return([true, ""])
+    allow_any_instance_of(MiqWorker::Runner).to receive(:worker_initialization)
+    allow_any_instance_of(MiqWorker::Runner).to receive(:worker_settings).and_return(settings)
```

Comment:
I remember @jrafanie and @chrisarcand saying `allow_any_instance_of` should be avoided during their summit presentation. Can this be avoided here?

This PR was referenced in their slides: https://github.com/rspec/rspec-mocks/pull/1060